In [26]:
import pandas as pd

In [27]:
df = pd.read_csv('../data/raw/customer_shopping_data.csv')

In [28]:
df.head()

,invoice_no,customer_id,gender,age,category,quantity,price,payment_method,invoice_date,shopping_mall
0,I138884,C241288,Female,28,Clothing,5,1500.40,Credit Card,5/8/2022,Kanyon
1,I317333,C111565,Male,21,Shoes,3,1800.51,Debit Card,12/12/2021,Forum Istanbul
2,I127801,C266599,Male,20,Clothing,1,300.08,Cash,9/11/2021,Metrocity
3,I173702,C988172,Female,66,Shoes,5,3000.85,Credit Card,16/05/2021,Metropol AVM
4,I337046,C189076,Female,53,Books,4,60.60,Cash,24/10/2021,Kanyon


In [29]:
df.shape

(99457, 10)

In [30]:
df.columns

Index(['invoice_no', 'customer_id', 'gender', 'age', 'category', 'quantity',
       'price', 'payment_method', 'invoice_date', 'shopping_mall'],
      dtype='object')

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99457 entries, 0 to 99456
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_no      99457 non-null  object 
 1   customer_id     99457 non-null  object 
 2   gender          99457 non-null  object 
 3   age             99457 non-null  int64  
 4   category        99457 non-null  object 
 5   quantity        99457 non-null  int64  
 6   price           99457 non-null  float64
 7   payment_method  99457 non-null  object 
 8   invoice_date    99457 non-null  object 
 9   shopping_mall   99457 non-null  object 
dtypes: float64(1), int64(2), object(7)
memory usage: 7.6+ MB


In [32]:
df.describe()

,age,quantity,price
count,99457.000000,99457.000000,99457.000000
mean,43.427089,3.003429,689.256321
std,14.990054,1.413025,941.184567
min,18.000000,1.000000,5.230000
25%,30.000000,2.000000,45.450000
50%,43.000000,3.000000,203.300000
75%,56.000000,4.000000,1200.320000
max,69.000000,5.000000,5250.000000


In [33]:
df['gender'].value_counts()

gender
Female    59482
Male      39975
Name: count, dtype: int64

In [34]:
df['category'].value_counts()

category
Clothing           34487
Cosmetics          15097
Food & Beverage    14776
Toys               10087
Shoes              10034
Souvenir            4999
Technology          4996
Books               4981
Name: count, dtype: int64

In [35]:
df['payment_method'].value_counts()

payment_method
Cash           44447
Credit Card    34931
Debit Card     20079
Name: count, dtype: int64

In [36]:
df['shopping_mall'].value_counts()

shopping_mall
Mall of Istanbul     19943
Kanyon               19823
Metrocity            15011
Metropol AVM         10161
Istinye Park          9781
Zorlu Center          5075
Cevahir AVM           4991
Forum Istanbul        4947
Viaport Outlet        4914
Emaar Square Mall     4811
Name: count, dtype: int64

In [37]:
NUMERICAL_COLUMNS = ["age", "quantity", "price"]

CATEGORICAL_COLUMNS = [
    "gender",
    "category",
    "payment_method",
    "shopping_mall"
]

IDENTIFIER_COLUMNS = [
    "invoice_no",
    "customer_id"
]

DATE_COLUMNS = ["invoice_date"]

OBSERVATIONS:

• The dataset contains more female than male transactions, with females accounting for approximately 60% of transactions and males accounting for approximately 40%.

• Clothing, Cosmetics, and Food & Beverage are the three most common purchase categories.

• Cash is the most frequently used payment method, while debit cards are the least frequently used.

• Mall of Istanbul, Kanyon, and Metrocity have the highest number of transactions among the shopping malls in the dataset.

• Customer ages range from 18 to 69, with an average age of approximately 43 years.

• The average purchase price is $689.26, while the median is $203.30. The difference between these values suggests that higher-priced purchases may be influencing the average.

• Customer IDs and invoice numbers appear to function primarily as identifiers rather than variables for numerical analysis.

• Invoice dates may be useful for analyzing purchasing trends over time.

• Further analysis could determine whether purchasing categories and shopping mall preferences differ by gender.